### ISSUE TITLE
add unit tests

### ISSUE THREAD

**Issue by abhijithvijayan** [abhijithvijayan/stargazed#19]
## Checklist
Add unit tests for the commands offered by the cli

**Comment by JacobMGEvans** [comment]
Could I get assigned to this?

**Comment by abhijithvijayan** [comment]
@JacobMGEvans sure

**Comment by abhijithvijayan** [comment]
@JacobMGEvans How is that PR coming ?

**Comment by JacobMGEvans** [comment]
I was able to get Jest working with some Babel, the tests themselves I am
still working on since I have two other projects I am working on right now
too. :)

**Comment by JacobMGEvans** [comment]
#22 will close issue when finished.

**Comment by JacobMGEvans** [comment]
I did add exports to a lot of the functions in the file, but those can easily be removed when determined which ones are not needed for isolated testing. 

I was also trying to get a feel for the codebase/the flow of data and how to approach testing it.

**Comment by abhijithvijayan** [comment]
@JacobMGEvans I think a lot of code refactoring is needed here and there in the source code. This was all written very soon so didn't get time to refactor.

**Comment by JacobMGEvans** [comment]
Well, tests will definitely help when you go to refactor because you can just run watch on them and make sure the tests don't break :) 

I try to write tests so the code implementation doesn't matter i.e. refactors dont break them... the only thing that matters is the input => output and behaviors remain the same/consistent.

**Comment by abhijithvijayan** [comment]
@JacobMGEvans You want me to assign this to someone else?

**Comment by JacobMGEvans** [comment]
Wow! Congratulations, definitely a good idea to refactor for performance then! 

You can assign other people, I probably will still write tests soon regardless. I am just finishing up with another issue from a different OSS and a few from my own side project.

**Comment by JacobMGEvans** [comment]
Also I really like your project and how you have communicated, it's been a pleasant experience. So I am certainly inclined to help 😄

**Comment by abhijithvijayan** [comment]
@JacobMGEvans Thanks a lot for that.

It's not everyday someone helps in such tiny OSS projects. Glad that you are ready to contribute. Appreciate it.

**Comment by JacobMGEvans** [comment]
Last night I started focusing on the unit tests, I am having issues importing the functions and statements for some reason... Might need to spend a little more time on the jest configuration, not sure what's the exact problem yet.

**Comment by abhijithvijayan** [comment]
@JacobMGEvans Told ya, it's messed up.

**Comment by JacobMGEvans** [comment]
lmao, I am certain I can figure it out I just wanted to keep you updated 😆

**Comment by JacobMGEvans** [comment]
Alright, I got this mostly figure out it was not configuration issues. The fact you were using a module.exports on an anonymous async function was making is impossible to test... I have some solutions, 
1. firstly I am going to make the test work with the code as-is then make the PR
2. I am going to refactor it to use ES6 modules `import/export` which export default should would the same as how you have `module.exports` with the testing in place we can refactor with confidence it will keep working as expected. 
3. Once the import/export is in place testing specific functions will be far easier and refactoring the code base will be far easier. 

4. Integration and E2E(likely not needed) will be the next endeavor

**Comment by abhijithvijayan** [comment]
@JacobMGEvans I am open to any modifications as long as the project works and code quality is improved.

**Comment by JacobMGEvans** [comment]
@abhijithvijayan Interesting enough if I try to pass false into the options that are primitives is throws an error, if I pass true it works fine... I might open a bug issue a little bit

**Comment by JacobMGEvans** [comment]
Would you be able to give me some mock or actual payloads and responses? I would rather not go through the whole process if you have anything like that leftover from developing it.

**Comment by JacobMGEvans** [comment]
Huh, it was working fine when I testing it locally...

Glad you got it fixed though Ill check out the changes and continue with testing.

**Comment by JacobMGEvans** [comment]
Someone also uncommented out tests that I had TODOs on because they were incomplete and a comment was added that "it doesnt return anything" which is why there was a TODO there to finish the test...

**Comment by abhijithvijayan** [comment]
@JacobMGEvans I was trying out the error fixes. Feel free to work on it.

**Comment by abhijithvijayan** [comment]
@JacobMGEvans Is babel config and its packages needed to run the tests?

**Comment by JacobMGEvans** [comment]
Yes.

**Comment by JacobMGEvans** [comment]
https://jestjs.io/docs/en/getting-started#using-babel

**Comment by abhijithvijayan** [comment]
@JacobMGEvans The tests are almost complete right!

**Comment by JacobMGEvans** [comment]
So far it's 50% coverage. Some of the functions and behavior are difficult
to test due to how errors are handled and also readme is written locally...
So that can overwrite the local repos readme when running parts of the code
in tests.

**Comment by abhijithvijayan** [comment]
@JacobMGEvans not really.

CLI creates README.md, and local file is readme.md

If you have tried the cli within the folder, you can see it doesn't affect the original readme.md file

**Comment by JacobMGEvans** [comment]
I wasn't asking a question I was stating what happens when you run the function in the Jest test, also it is not cap sensitive, Node just writes to the readme.md, like I said.

**Comment by abhijithvijayan** [comment]
I thought since the command `Node cli.js -u "USERNAME"` creates `README.md` along with `readme.md`, it wouldnt be a problem too, while testing.


Maybe renaming current `readme.md` to just `readme` might help right!

**Comment by JacobMGEvans** [comment]
Perhaps. I haven't been able to get to it for a few days.

**Comment by abhijithvijayan** [comment]
getting stale?

**Comment by abhijithvijayan** [comment]
@JacobMGEvans A lot has changes since the last test added.

I have refactored the code a lot and it seems to be stable now. All the test suites are still working as expected. 

Are you currently on other projects now for not be able to work on this issue anymore?

**Comment by JacobMGEvans** [comment]
Yeah, it is actually my work that has me swamped... Sorry.
